airbyte-integrations/connectors/source-db2/src/test/java/io.airbyte.integrations.source.db2/Db2TestDatabase.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.db2;
+
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.cdk.db.factory.DatabaseDriver;
+import io.airbyte.cdk.db.jdbc.JdbcUtils;
+import io.airbyte.cdk.testutils.TestDatabase;
+import io.airbyte.commons.json.Jsons;
+import java.util.stream.Stream;
+import org.jooq.SQLDialect;
+import org.testcontainers.containers.Db2Container;
+
+public class Db2TestDatabase extends
+    TestDatabase<Db2Container, Db2TestDatabase, Db2TestDatabase.Db2DbConfigBuilder> {
+
+  private static boolean containerStarted = false;
+  private final Db2Container container;
+
+  protected Db2TestDatabase(final Db2Container container) {
+    super(container);
+    this.container = container;
+  }
+
+  @Override
+  public Db2TestDatabase initialized() {
+    if (!containerStarted) {
+      container.start();
+      containerStarted = true;
```

Comment:
The `.start()` method is idempotent so `containerStarted` isn't needed. Plus, it's static for some reason.

Reply:
@postamar The `start` method is called from `createTestDatabase` within `setup` which is executed before each unit test,  I wanted to use the same connector for all the unit tests and not have docker container start and stop again. In order to do that I had to introduce `containerStarted` and make it static
```
  @BeforeEach
  public void setup() throws Exception {
    customSetup();
    testdb = createTestDatabase();
```

Reply:
I understand. The place to put the `.start()` call in the `@BeforeAll` static method.